EU4toV2/Source/Mappers/CultureGroups/CultureGroups.cpp

```diff
@@ -40,35 +50,75 @@ void mappers::CultureGroups::registerKeys()
 		{
 			std::vector<Culture> cultures;
 			CultureGroup newGroup(cultureGroupName, theStream);
-			for (const auto& culture: newGroup.getCultures())
+			if (cultureGroupsMap.count(cultureGroupName))
 			{
-				cultureToGroupMap.insert(make_pair(culture.first, newGroup));
-				cultures.push_back(culture.second);
-			}
-
-			const auto& itr = groupToCulturesMap.find(cultureGroupName);
-			if (itr != groupToCulturesMap.end())
-			{
-				const auto& oldCultures = itr->second;
-				for (const auto& oldCulture: oldCultures)
+				// We would normally override base definitions with incoming CK2 ones, but CK2 definitions
+				// are crap and don't actually list all required cultures, so we have to merge.
+				for (const auto& cultureItr: newGroup.getCultures())
 				{
-					cultures.push_back(oldCulture);
-				}
+					cultureGroupsMap[cultureGroupName].mergeCulture(cultureItr.first, cultureItr.second);
+				}				
```

Comment:
nit: whitespace randomly inserted at the end of line 60 here.

Reply:
dropped.